### ISSUE TITLE
`LT` instance for Option is unused/not useful to have at this kind of base level.

### ISSUE THREAD

**Issue by linesthatinterlace** [leanprover/lean4#3100]
### Prerequisites

* [X ] Put an X between the brackets on this line if you have done all of the following:
    * Check that your issue is not already [filed](https://github.com/leanprover/lean4/issues).
    * Reduce the issue to a minimal, self-contained, reproducible test case. Avoid dependencies to mathlib4 or std4.

### Description

The file Init/Data/Option/Basic.lean contains a function `Option.lt`, an instance of LT for `Option A` inherited from a `LT A`, and a decidability instance for that relation.

The definition that it uses is one which sets `none < some a` and `some a < some b` just when `a < b`. This is a perfectly sane definition, but making this choice at such a low-level seems counter-intuitive to me. Different libraries might want to choose different definitions for LT on Option. Back in Mathlib 4 land, `WitBot A` and `WithTop A` are synonyms for `Option` that are based off this ordering and the obvious dual one. The base `Option` type having a LT instance as a core feature of `lean4` - when as far as I can see this isn't used anywhere - removes the ability of downstream projects like `std4` or `mathlib4` to make different choices.

### Context

I was doing some work in Mathlib4 and I used `<` in some statements about Option, It worked (because the instance is defined), and some stuff is RFL-true and so it wasn't until substantially later that I realised that this instance wasn't particular doing what I had expected and didn't have any nice API. I discussed this in a Zulip thread here: https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Preorder.20on.20Option

I think downstream in Mathlib4 (or maybe std4 but I am not as sure how to contribute to that) we should add an instance, but in the context of this project I think it should simply be removed - it isn't useful and it creates footguns like what happened to me.

### Steps to Reproduce

1. Write down the following code:
``` 
example {A : Type*} [LT A] {a : A} : (none : Option A) < some a := sorry` 
```
2. The code is trivially true (I think `trivial` works as a proof through I am not sure.) 
3. And this typechecks.

**Expected behavior:**
I would expect the default behaviour of LT on Option to be either:
a) LT is not defined, and so this shouldn't typecheck/synthesise a correct instance.
b) An instance is defined, and in it, this is not a true statement. Why should `none < some a` be true, and `some a < none` not? You don't have to make a choice.

**Actual behavior:** 
A choice is made: this statement is true.

### Versions

4.30
Windows 11 running Ubuntu WSL2

### Additional Information

None.

### Impact

Add :+1: to [issues you consider important](https://github.com/leanprover/lean4/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc). If others are impacted by this issue, please ask them to add :+1: to it.

**Comment by linesthatinterlace** [comment]
I have created a draft PR here, mainly to experiment and see if this breaks anything upstream by just removing it. That is #3102.

**Comment by eric-wieser** [comment]
It seems that `lt` was turned into an instance in https://github.com/leanprover/lean4/commit/75a67ceb295a46a0125862e223566f7f5f6e27a4.

**Comment by linesthatinterlace** [comment]
Indeed, in a sense I'm proposing directly reverting that commit.

**Comment by linesthatinterlace** [comment]
Happily, #3102 seems to not break anything! So we lose literally nothing by doing this. There is a question of where an appropriate place for Option instances then is - my gut says that it should be in std4 for the basic instances and mathlib4 for anything in the actual order hierarchy? But ideally this would be resolved first.

**Comment by eric-wieser** [comment]
> Indeed, in a sense I'm proposing directly reverting that commit.

I don't think you are; reverting that commit would mean downgrading the `instance` to a `def`. You're proposing the other reasonable thing of deleting it entirely!

**Comment by semorrison** [comment]
I'm personally in favour. It's great to see that the Mathlib CI shows this isn't being used. I agree this is the sort of thing where the choice should be made via instances for synonyms. (Perhaps someone needs an `IncomparableNone` synonym? Great!)

**Comment by nomeata** [comment]
NB that there is more discussion in https://github.com/leanprover/lean4/pull/3102#issuecomment-1865876110.

**Comment by linesthatinterlace** [comment]
Also in the Zulip thread linked above, which is probably a better place than any individual PR for hashing out details.

**Comment by linesthatinterlace** [comment]
(While this issue pertains only to core, there's discussions that span the up-downstream spectrum.)

**Comment by nomeata** [comment]
> Also in the Zulip thread linked above, which is probably a better place than any individual PR for hashing out details.

Thanks for the pointer. I see that Mario is arguing as I do, so I think it’s well taken care of :)

**Comment by linesthatinterlace** [comment]
I agree, but Zulip is probably the appropriate place for discussing the latter too.
